Enable Python 3.6

jakirkham opened the pull request:
Fixes https://github.com/conda-forge/pycrypto-feedstock/issues/5
Closes https://github.com/conda-forge/pycrypto-feedstock/pull/3

* Enables Python 3.6 builds of pycrypto.
* Adds a hack to pick up the MSVC provided `stdint.h` if available.

Inspired by this [suggestion]( https://github.com/dlitz/pycrypto/issues/218#issuecomment-304484949 ).

jakirkham committed:
Drop Python 3.6 skip

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.3.1

Enable Python 3.6 builds of pycrypto.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Force use of MSVC's stdint.h when available

If the MSVC version used has `stdint.h` (namely VS 2015+), then force
use of the MSVC provided `stdint.h` instead of using the stand-in copy
of `stdint.h` provided by `pycrypto`. This should fix some build issues
encountered with Python 3.6 and allow for a more proper build on Python
3.5 when using MSVC.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -23,10 +23,10 @@ source:
+  number: 1
   script:
-    - export LDFLAGS="-headerpad_max_install_names"  # [osx]
+    - set CL="/FI %VCINSTALLDIR%\\include\\stdint.h %CL%"  # [win and py>=35]
```
This [forces]( https://msdn.microsoft.com/en-us/library/8c5ztk84.aspx ) a particular copy of `stdint.h` to be used by all `#include` directives in the C code. Should note this happens in spite of the `stdint.h` that PyCrypto [includes]( https://github.com/dlitz/pycrypto/blob/v2.6.1/src/inc-msvc/stdint.h ).

jakirkham committed:
Bump build number to 1

jakirkham commented on recipe/meta.yaml:
```diff
@@ -23,10 +23,10 @@ source:
+  number: 1
   script:
-    - export LDFLAGS="-headerpad_max_install_names"  # [osx]
+    - set CL=/FI"%VCINSTALLDIR%\\INCLUDE\\stdint.h" %CL%  # [win and py>=35]
```
This [forces]( https://msdn.microsoft.com/en-us/library/8c5ztk84.aspx ) a particular copy of `stdint.h` to be used by all `#include` directives in the C code. Should note this overrides the `stdint.h` that PyCrypto [provides]( https://github.com/dlitz/pycrypto/blob/v2.6.1/src/inc-msvc/stdint.h ).

jakirkham commented:
cc @lexual
